File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        if(s.isEmpty()){
            return true;
        }
        int first = 0;
        int last = s.length()-1;
        while(first <= last){
            char currF = s.charAt(first);
            char currl = s.charAt(last);
            if( !Character.isLetterOrDigit(currF) ){
                first ++;
            } else if( !Character.isLetterOrDigit(currl) ){
                last --;
            } else{
                if(Character.toLowerCase(currF) != Character.toLowerCase(currl)){
                    return false;
                }
                first++;
                last--;
            }
        }
        return true;

    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s1 = ""
        for ch in s:
            if ch.isalnum():
                s1+=ch.lower()
        left = 0
        right = len(s1) - 1

        while left < right:
            if s1[left] != s1[right]:
                return False
            left += 1
            right -= 1

        return True